src/middlewared/middlewared/plugins/kubernetes_linux/lifecycle.py

```diff
@@ -160,6 +160,22 @@ async def post_start_internal(self):
             ]
         )
 
+        # Wait for taints to be removed - this includes the built-in taints which might be present
+        # at this point i.e node.kubernetes.io/not-ready
+        taint_timeout = 600
+        taints = None
+        while taint_timeout > 0:
+            taints = (await self.middleware.call('k8s.node.config'))['spec'].get('taints')
+            if not taints:
+                break
+
+            await asyncio.sleep(5)
```

Comment:
We should create another variable for sleep time `sleep_time` or something equivalent. Also, should we wait 5 seconds each iteration?

Reply:
We can change it to something else but 5 seconds looks like a decent delay?
In most cases we should not be hitting these sleeps at all..